features/steps.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+from contextlib import closing
+import datetime
+import os
+from journal import INSERT_ENTRY
+from journal import connect_db
+from journal import DB_SCHEMA
+from lettuce import world
+from lettuce import step
+from lettuce import before
+from lettuce import after
+
+TEST_DSN = 'dbname=test_learning_journal user=nbeck'
+
+
+def init_db(setup):
+    with closing(connect_db(setup)) as db:
+        db.cursor().execute(DB_SCHEMA)
+        db.commit()
+
+
+def empty_db(setup):
+    with closing(connect_db(setup)) as db:
+        db.cursor().execute("DROP TABLE entries")
+        db.commit()
+
+
+def clear_entries(setup):
+    with closing(connect_db(setup)) as db:
+        db.cursor().execute("DELETE FROM entries")
+        db.commit()
+
+
+@before.all
+def db():
+    setup = {'db': TEST_DSN}
+    init_db(setup)
+    world.setup = setup
+
+
+@before.all
+def app():
+    from journal import main
+    from webtest import TestApp
+    os.environ['DATABASE_URL'] = TEST_DSN
+    app = main()
+    world.app = TestApp(app)
+
+
+def run_query(db, query, params=(), get_results=True):
+    cursor = db.cursor()
+    cursor.execute(query, params)
+    db.commit()
+    results = None
+    if get_results:
+        results = cursor.fetchall()
+    return results
+
+
+@before.all
+def login():
+    entry_data = {
+        'username': 'admin',
+        'password': 'secret',
+    }
+    world.app.post('/login', params=entry_data, status='3*')
+
+
+@before.all
+def entry1():
+    setup = world.setup
+    now = datetime.datetime.utcnow()
+    expected = ('New Title', 'New Test', now)
+    with closing(connect_db(setup)) as db:
+        run_query(db, INSERT_ENTRY, expected, False)
+        db.commit()
+    world.expected = expected
+
+
+@before.all
+def entry2():
+    """Add an entry to the database with a header"""
+    setup = world.setup
+    now = datetime.datetime.utcnow()
+    expected = ('Markdown Test', '# Header1', now)
+    with closing(connect_db(setup)) as db:
+        run_query(db, INSERT_ENTRY, expected, False)
+        db.commit()
+    world.expected = expected
+
+
+@before.all
+def entry3():
+    """Add an entry to the database with a header"""
+    setup = world.setup
+    now = datetime.datetime.utcnow()
+    expected = ('Markdown Test', "```Inline code```", now)
+    with closing(connect_db(setup)) as db:
+        run_query(db, INSERT_ENTRY, expected, False)
+        db.commit()
+    world.expected = expected
+
+
+@after.all
+def cleanup(step):
+    empty_db(world.setup)
+
+
+@step('I am at the homepage')
+def at_home(step):
+    world.url = "http://127.0..0.0.1:5000/"
```

Comment:
It seems like just saying that the url is the homepage url doesn't in any meaningful sense mean that you are "at the homepage".  I would go ahead and load this url via `world.app.get(url)`.